numpy/_core/src/umath/ufunc_object.c

```diff
@@ -1156,6 +1156,14 @@ prepare_ufunc_output(PyUFuncObject *ufunc,
         PyArrayObject *arr;
         PyObject *args_tup;
 
+        /* Use a deprecation warning, since end-users shouldn't worry. */
+        if (DEPRECATE(
```

Comment:
Hmm, this seems a bit strange: you deprecate the call out, but any subclass that does `super().__array_prepare__(...)` will now fail. Or am I missing something?